dpnp/backend/kernels/dpnp_krnl_random.cpp

```diff
@@ -926,6 +927,75 @@ void dpnp_rng_rayleigh_c(void* result, const _DataType scale, const size_t size)
     }
 }
 
+template <typename _DataType>
+void dpnp_rng_shuffle_c(
+    void* result, const size_t itemsize, const size_t ndim, const size_t high_dim_size, const size_t size)
+{
+    if (!(size) || !(high_dim_size > 1))
+    {
+        return;
+    }
+
+    char* result1 = reinterpret_cast<char*>(result);
+
+    double* Uvec = nullptr;
+
+    size_t uvec_size = high_dim_size - 1;
+    Uvec = reinterpret_cast<double*>(dpnp_memory_alloc_c(uvec_size * sizeof(double)));
+    mkl_rng::uniform<double> uniform_distribution(0.0, 1.0);
+    auto uniform_event = mkl_rng::generate(uniform_distribution, DPNP_RNG_ENGINE, uvec_size, Uvec);
+    uniform_event.wait();
+
+    if (ndim == 1)
+    {
+        // Fast, statically typed path: shuffle the underlying buffer.
+        // Only for non-empty, 1d objects of class ndarray (subclasses such
+        // as MaskedArrays may not support this approach).
+        // TODO
+        // kernel
+        char* buf = nullptr;
+        buf = reinterpret_cast<char*>(dpnp_memory_alloc_c(itemsize * sizeof(char)));
+        for (size_t i = uvec_size; i > 0; i--)
+        {
+            size_t j = (size_t)(floor((i + 1) * Uvec[i - 1]));
+            memcpy(buf, result1 + j * itemsize, itemsize);
```

Comment:
I do not understand the overall idea of this code.
1. create temp buffer
2. copy something to this buffer
3. free temp buffer

It is no sense, isn't it?

Reply:
?